src/reflect/value.go

```diff
@@ -378,7 +378,22 @@ func (v Value) String() string {
 }
 
 func (v Value) Bytes() []byte {
-	panic("unimplemented: (reflect.Value).Bytes()")
+	switch v.Kind() {
+	case Slice:
+		if v.typecode.elem().Kind() != Uint8 {
+			panic(&ValueError{Method: "Bytes", Kind: v.Kind()})
+		}
+		return *(*[]byte)(v.value)
+
+	case Array:
+		if v.typecode.elem().Kind() != Uint8 {
+			panic(&ValueError{Method: "Bytes", Kind: v.Kind()})
+		}
+
+		return unsafe.Slice((*byte)(v.value), v.Len())
```

Comment:
Actually I realize this code still isn't correct. When run under the regular Go compiler, it panics:

```
panic: reflect.Value.Bytes of unaddressable byte array

goroutine 1 [running]:
reflect.Value.bytesSlow({0x1022ee160?, 0x14000018080?, 0x14000044768?})
	/Users/ayke/lib/go/src/reflect/value.go:321 +0xec
reflect.Value.Bytes(...)
	/Users/ayke/lib/go/src/reflect/value.go:305
main.main()
	/Users/ayke/src/tinygo/tinygo/tmp/smallslice.go:7 +0x108
exit status 2
```

Which makes sense: this would allow the caller to modify the array in-place. Also see the documentation (emphasis mine):

> It panics if v's underlying value is not a slice of bytes or an **addressable array of bytes**.

So in short, it should panic for any array that is not addressable (I assume this means `CanAddr`). This conveniently also fixes the issue with very short arrays that fit in a pointer.

Reply:
Special casing removed and I added a comment.